secure_message/resources/health.py

```diff
@@ -0,0 +1,57 @@
+import logging
+
+from flask_restful import Resource
+from flask import jsonify, current_app
+from structlog import wrap_logger
+
+from secure_message.repository.retriever import Retriever
+from secure_message.services.service_toggles import party
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+class Health(Resource):
+
+    """Rest endpoint to provide application general health"""
+
+    @staticmethod
+    def get():
+        return "{status : healthy}"
+
+
+class DatabaseHealth(Resource):
+
+    """Rest endpoint to provide application database health"""
+
+    @staticmethod
+    def get():
+        return Retriever().check_db_connection()
+
+
+class HealthDetails(Resource):
+
+    """Rest endpoint to provide application details"""
+
+    @staticmethod
+    def get():
+        """returns environment and api endpoint details"""
+        func_list = {}
+        for rule in current_app.url_map.iter_rules():
+            if rule.endpoint != 'static':
+                func_list[rule.rule] = current_app.view_functions[rule.endpoint].__doc__
+
+        details = {'Name': current_app.config['NAME'],
+                   'Version': current_app.config['VERSION'],
+                   'SMS Log level': current_app.config['SMS_LOG_LEVEL'],
+                   'APP Log Level': current_app.config['APP_LOG_LEVEL'],
+                   'Database URL': current_app.config['SQLALCHEMY_DATABASE_URI'],
```

Comment:
Should we be exposing Database URL in health checks as it contains the DB credentials

Reply:
Again, beyond the scope of this PR but good to flag for consideration.